### PR TITLE
Bugfix and updated parameter layout

### PR DESCRIPTION
**Pull request by mehditlili**

### CODE REVIEW

File: turtlebot_navigation/launch/includes/move_base.launch.xml [hunk 1]
```diff
@@ -20,11 +20,11 @@
     <rosparam file="$(find turtlebot_navigation)/param/move_base_params.yaml" command="load" />
     
     <!-- reset frame_id parameters using user input data -->
-    <rosparam name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
-    <rosparam name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
-    <rosparam name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
-    <rosparam name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
-    <rosparam name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>
```
Comment (stonier) [inline turtlebot_navigation/launch/includes/move_base.launch.xml:27]: This is very curious...how was this previously working? The [rosparam format](http://wiki.ros.org/roslaunch/XML/rosparam) doesn't have a 'value' tag.

Ah, this is a bugfix for 6a092016b95cb3555a5cdc7af111317671906180.

File: turtlebot_navigation/param/move_base_params.yaml [hunk 2]
```diff
@@ -15,3 +15,37 @@ oscillation_distance: 0.2
 
 # local planner - default is trajectory rollout
 base_local_planner: "dwa_local_planner/DWAPlannerROS"
+
+## recovery behaviors; we avoid spinning, but we need a fall-back replanning
```
Comment (stonier) [inline turtlebot_navigation/param/move_base_params.yaml:20]: Might like to add some comments about why these commented parameters are here.

File: turtlebot_navigation/param/costmap_common_params.yaml [hunk 3]
```diff
@@ -5,17 +5,57 @@ raytrace_range: 3.0
 # Obstacle Cost Shaping (http://wiki.ros.org/costmap_2d/hydro/inflation)
 robot_radius: 0.20  # distance a circular robot should be clear of the obstacle (kobuki: 0.18)
 # footprint: [[x0, y0], [x1, y1], ... [xn, yn]]  # if the robot is not circular
-inflation_radius: 0.50  # max. distance from an obstacle at which costs are incurred for planning paths.
-cost_scaling_factor: 5  # exponential rate at which the obstacle cost drops off (default: 10)
+#inflation_radius: 0.50  # max. distance from an obstacle at which costs are incurred for planning paths.
```
Comment (stonier) [inline turtlebot_navigation/param/costmap_common_params.yaml]: Note - these are moved down to the obstacle_layer parameters. Let's delete them, they just clutter.

File: turtlebot_navigation/param/costmap_common_params.yaml [hunk 4]
```diff
@@ -5,17 +5,57 @@ raytrace_range: 3.0
...
+#cost_scaling_factor: 5  # exponential rate at which the obstacle cost drops off (default: 10)
 
 # voxel map configuration; z-voxels 0 are filled by bumpers and 1 by laser scan (kinect)
-map_type: voxel
-origin_z: 0.0
-z_resolution: 0.2
-z_voxels: 2
-publish_voxel_map: false
+#map_type: voxel
```
Comment (stonier) [inline turtlebot_navigation/param/costmap_common_params.yaml]: Delete these commented variables, they just create clutter.

**Comment by mehditlili** [comment]
The new plugin layout needs further testing, I will probably commit some other changes.

**Comment by stonier** [comment]
ok

**Comment by mehditlili** [comment]
It works well now.

**Comment by mehditlili** [comment]
In turtlebot_navigation/param/costmap_common_params.yaml line 30 and 31, min_obstacle_height and max_obstacle_height should be changed if you move the 3d sensor.

**Comment by jihoonl** [comment]
:+1:

**Pull request merged**